src/lib/Microsoft.Health.Extensions.Fhir.R4/Search/SearchExtensions.cs

```diff
@@ -197,5 +198,51 @@ public static string ToSearchQueryParameter(this Hl7.Fhir.Model.Identifier ident
 
             return $"identifier={identifier.ToSearchToken()}";
         }
+
+        public static string ToExpandedSearchQueryParameters(this Hl7.Fhir.Model.Identifier identifier)
+        {
+            EnsureArg.IsNotNull(identifier?.Value, nameof(identifier.Value));
+
+            string[] components = identifier.Value?.Split('.');
+
+            // Check if the identifier contains at least 5 components separated by '.', e.g. 'patientId.deviceId.typeName.startTime.endTime'.
+            if (components.Length >= 5)
+            {
+                // Make sure the start component can be converted to an ISO 8601 time string.
+                if (TryGetFormattedDateString(components[components.Count() - 2], out string start))

```

Comment:
nit: Just curious, are we not collapsing these 'ifs' with '&&' for the individual inline comments?
(Fine with this btw was just curious :) )

Reply:
This was just a code style choice that I made to separate and comment each component. I just liked the idea of handling each identifier component separately and commenting each one separately.